### PR TITLE
RFC: GitHub Actions

### PR DESCRIPTION
**Pull request by CrossNox**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+jobs:
+  run_tests:
+    runs-on: python:3.8
+    steps:
+    - name: Update base image
+      run: |
+        apt update
+        apt install -y libkrb5-dev libsasl2-dev
+        pip install poetry
```
Comment (MartinCura) [inline .github/workflows/test.yml:12]: make sure to specify the `poetry` version, e.g. `poetry==1.1.13`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+name: test
+on: [pull_request, push]
+
+jobs:
+  run_tests:
+    runs-on: python:3.8
```
Comment (MartinCura) [inline .github/workflows/test.yml:6]: Maybe you want
```suggestion
    runs-on: [self-hosted]
    container: python:3.8
```
on all of them

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    runs-on: self-hosted
+    steps:
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+    - name: Update base image
+      run: |
+        apt update
+        apt install -y libkrb5-dev libsasl2-dev
```
Comment (MartinCura) [inline .github/workflows/test.yml:15]:
```suggestion
        sudo apt update
        sudo apt install -qy libkrb5-dev libsasl2-dev
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      run: |
+        poetry install -E gsheets -vvv
+        poetry run python -c 'from muttlib import gsheetsconn'
+    - name: Test gdrive
+      run: |
+        poetry install -E gdrive -vvv
+        poetry run python -c 'from muttlib import gdrive'
+    - name: Run nox
+      run: |
+        poetry run nox --envdir $GITHUB_WORKSPACE/.nox --sessions tests -- -n 2
```
Comment (MartinCura) [inline .github/workflows/test.yml:36]:
```suggestion
        poetry run nox --envdir $GITHUB_WORKSPACE/.nox --sessions tests
```

**Comment by JDSanto** [comment]
We only need to change `validate.yml` to run on master branch.

@MartinCura @CrossNox feel free to check PR

File: .bumpversion.cfg [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.4.19
+current_version = 1.4.20.alpha
```
Comment (MartinCura) [inline .bumpversion.cfg]: 👀

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
 on: [pull_request, push]
 
+env:
+  MIN_LINE_RATE: 0.2
+
 jobs:
   run_tests:
     runs-on: self-hosted
+    container:
+      image: python:3.8
```
Comment (MartinCura) [inline .github/workflows/test.yml:11]: change these to `python:3.10`? see https://github.com/MuttData/muttlib/pull/6

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.4.20] - 2022-11-03
+### Changed
+- Migrated GitLab CI/CD to Github Action
```
Comment (MartinCura) [inline CHANGELOG.md:10]:
```suggestion
- Migrated GitLab CI/CD to Github Actions
```

File: .github/workflows/test.yml (hunk 6 above)
Comment (JDSanto) [reply .github/workflows/test.yml:11]: just in case let's do this in that PR

**Pull request merged**